### PR TITLE
Move the GetResources logic to PodContainerManager

### PR DESCRIPTION
**Pull request by miaoyq**
**What this PR does / why we need it**:
Move the GetResources logic from `ContainerManager` to `PodContainerManager`
Address a TODO: `TODO: move the GetResources logic to PodContainerManager.`, and related to this [comment](https://github.com/kubernetes/kubernetes/pull/51209/files#r135586110)
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by feiskyer** [comment]
@miaoyq Could add an explation of why introducing this change?

**Comment by miaoyq** [comment]
@feiskyer  See: https://github.com/kubernetes/kubernetes/pull/51209/files#r135586110

**Comment by miaoyq** [comment]
@feiskyer  I will add an expiation for this. :)

File: pkg/kubelet/cm/container_manager_linux.go [hunk 1]
```diff
@@ -307,9 +306,10 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 func (cm *containerManagerImpl) NewPodContainerManager() PodContainerManager {
 	if cm.NodeConfig.CgroupsPerQOS {
 		return &podContainerManagerImpl{
```
Comment (jiayingz) [inline pkg/kubelet/cm/container_manager_linux.go]: Does this mean deviceplugin feature would depend on the enabling of CgroupsPerQOS feature after this change? I would like to keep the two features separate as they don't seem to have any hard dependency.

Comment (tengqm) [reply pkg/kubelet/cm/container_manager_linux.go]: agree to jiayinz, these two should be kept independent from each other.

**Pull request closed without merging**